lib/online_order.rb

```diff
@@ -0,0 +1,122 @@
+require 'csv'
+require 'awesome_print'
+require_relative 'order.rb'
+
+
+#_______________ Grocery::OnlineOrder _______________
+
+module Grocery
+  class OnlineOrder < Grocery::Order
+
+    attr_reader :id, :products, :customer_id, :status
+
+    # Initialize class OnlineOrder:
+    def initialize(id, products, customer_id, status = :pending)
+      super(id, products)
+      @customer_id = customer_id.to_i
+      @status = status #pending, paid, processing, shipped or complete
+    end
+
+
+    #############################################################################################
+    # TOTAL  OF ORDER:
+
+    #  The total method should be the same, except it will add a $10 shipping fee
+    # it "Doesn't add a shipping fee if there are no products" do
+    def total
+
+      if @products != nil
+        return 10 + super
+      end
+
+    end
+
+
+    #############################################################################################
+    # ADD PRODUCT TO ORDER:
+
+    # The add_product method should be updated to permit a new product to be added ONLY if the status is either pending or paid (no other statuses permitted)
+    def add_product(product_name, product_price)
+
+      unless @status == :paid || @status == :pending
+        raise ArgumentError.new("Only paid or pending status is allowed.")
+      end
+      super(product_name, product_price)
+
+    end
+
+    #############################################################################################
+    # ALL ORDERS:
+
+    # self.all - returns a collection of OnlineOrder instances, representing all of the OnlineOrders described in the CSV. See below for the CSV file specifications
+    def self.all
+
+      @all_orders = []
+
+      # Read file:
+      read_file = CSV.read('../support/online_orders.csv', 'r')
+
+      read_file.each do |row| # CSV.read('../support/orders.csv', 'r').each do |row|
+        #Select the order id number from the file and assign it:
+        order_id = row[0].to_i
+        customer_id = row[2].to_i
+        status = row[3]
+
+        # Separete the elements after the first comma (index[1]) into {product_name, product_price} and assign it to a products variable:
+        products = separate("#{row[1]}".split(';'))
+        # example => puts products = [{"Allspice"=>"64.74"}, {"Bran"=>"14.72"}, {"UnbleachedFlour"=>"80.59"}]
+
+        # Push this order (order id, products(itens, price)) to the array of orders
+        new_order = OnlineOrder.new(order_id, products, customer_id, status)
+        @all_orders << new_order
+      end
+      return @all_orders
+    end
+
+
+    # Separete the elements into (product_name, product_price):
+    def self.separate(elements_of_order)
+      super  #(The same as in Grocery::Order)
+    end
+
+
+
+    #############################################################################################
+    # FIND ORDER:
+
+    def self.find(find_id)
```

Comment:
With inheritance if the method doesn't do anything different you can just **not** define a new copy and use the superclass' version.

So if you took this method out, it would still work and pass the tests.